app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt

```diff
@@ -0,0 +1,208 @@
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.platformparameter.PlatformParameterController
+import org.oppia.android.domain.platformparameter.PlatformParameterModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.OppiaTestRule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG_VALUE
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests to verify the working of Platform Parameter Architecture. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = PlatformParameterIntegrationTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class PlatformParameterIntegrationTest {
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var platformParameterController: PlatformParameterController
+
+  private val mockPlatformParameterList by lazy {
+    val mockSplashScreenWelcomeMsgParam = PlatformParameter.newBuilder()
+      .setName(SPLASH_SCREEN_WELCOME_MSG)
+      .setBoolean(SPLASH_SCREEN_WELCOME_MSG_VALUE)
+      .build()
+
+    listOf<PlatformParameter>(
+      mockSplashScreenWelcomeMsgParam
+    )
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  @Test
+  fun testSplashTestActivity_readEmptyDatabase_checkWelcomeMsgIsInvisibleByDefault() {
+    launch(SplashTestActivity::class.java).use {
+      onView(withText(SplashTestActivity.WELCOME_MSG)).check(doesNotExist())
+    }
+  }
+
+  @Test
+  fun testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible() {
```

Comment:
this test is failing for Gradle on robolectric.

Reply:
But these tests are running fine on my machine

Reply:
```
org.oppia.android.app.testing.PlatformParameterIntegrationTest > testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible FAILED
    androidx.test.espresso.NoMatchingRootException: Matcher 'is toast' did not match any of the following roots: [Root{application-window-token=android.view.ViewRootImpl$W@247955c1, window-token=android.view.ViewRootImpl$W@247955c1, has-window-focus=true, layout-params-type=1, layout-params-string={(0,0)(fillxfill) ty=BASE_APPLICATION wanim=0x10302f8
      fl=LAYOUT_IN_SCREEN LAYOUT_INSET_DECOR SPLIT_TOUCH HARDWARE_ACCELERATED DRAWS_SYSTEM_BAR_BACKGROUNDS
      pfl=FORCE_DRAW_STATUS_BAR_BACKGROUND}, decor-view-string=DecorView{id=-1, visibility=VISIBLE, width=960, height=1410, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params={(0,0)(fillxfill) ty=BASE_APPLICATION wanim=0x10302f8
      fl=LAYOUT_IN_SCREEN LAYOUT_INSET_DECOR SPLIT_TOUCH HARDWARE_ACCELERATED DRAWS_SYSTEM_BAR_BACKGROUNDS
      pfl=FORCE_DRAW_STATUS_BAR_BACKGROUND}, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=1}}]
        at java.base/java.lang.Thread.getStackTrace(Thread.java:1654)
        at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:96)
        at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
        at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:322)
        at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:306)
        at org.oppia.android.app.testing.PlatformParameterIntegrationTest.testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible(PlatformParameterIntegrationTest.kt:133)
```

1. Could you add a screenshot showing that this robolectric test is passing for you?
2. Could you confirm is the `getToastMatcher` function working fine?
3. COuld you add some reference if any, for the `getToastMatcher()` function?

Reply:
### 1 and 2 are proven by this screenshot 
![Screenshot 2021-07-07 121030](https://user-images.githubusercontent.com/64064110/124807884-e58b1180-df7b-11eb-9b00-7e3086e1f3f2.png)

### getToastMatcher( ) 
Answer 2 [Reference-SO](https://stackoverflow.com/questions/28390574/checking-toast-message-in-android-espresso)